src/main/java/com/gooddata/GoodData.java

```diff
@@ -179,6 +223,17 @@ private RestTemplate createRestTemplate(String login, String password, String ho
         return restTemplate;
     }
 
+    private void applySettings(final GoodDataSettings settings, final HttpClientBuilder clientBuilder) {
```

Comment:
please rename the method to `createHttpClientBuilder` and let it return the new instance